x/emissions/keeper/msgserver/msg_server_topics.go

```diff
@@ -5,11 +5,60 @@ import (
 
 	errorsmod "cosmossdk.io/errors"
 	alloraMath "github.com/allora-network/allora-chain/math"
-	"github.com/allora-network/allora-chain/x/emissions/types"
+	emissionstypes "github.com/allora-network/allora-chain/x/emissions/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-func (ms msgServer) CreateNewTopic(ctx context.Context, msg *types.MsgCreateNewTopic) (*types.MsgCreateNewTopicResponse, error) {
-	if err := msg.Validate(); err != nil {
+func validateNewTopic(msg *emissionstypes.MsgCreateNewTopic) error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+
+	if len(msg.LossMethod) == 0 {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "loss method cannot be empty")
+	}
+	if msg.EpochLength <= 0 {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "epoch length must be greater than zero")
+	}
+	if msg.WorkerSubmissionWindow == 0 {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "worker submission window must be greater than zero")
+	}
+	if msg.GroundTruthLag < msg.EpochLength {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "ground truth lag cannot be lower than epoch length")
+	}
+	if msg.WorkerSubmissionWindow > msg.EpochLength {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "worker submission window cannot be higher than epoch length")
+	}
+	if msg.AlphaRegret.Lte(alloraMath.ZeroDec()) ||
+		msg.AlphaRegret.Gt(alloraMath.OneDec()) ||
+		emissionstypes.ValidateDec(msg.AlphaRegret) != nil {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "alpha regret must be greater than 0 and less than or equal to 1")
+	}
+	if msg.PNorm.Lt(alloraMath.MustNewDecFromString("2.5")) ||
+		msg.PNorm.Gt(alloraMath.MustNewDecFromString("4.5")) ||
+		emissionstypes.ValidateDec(msg.PNorm) != nil {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "p-norm must be between 2.5 and 4.5")
+	}
+	if msg.Epsilon.Lte(alloraMath.ZeroDec()) || msg.Epsilon.IsNaN() || !msg.Epsilon.IsFinite() {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "epsilon must be greater than 0")
+	}
+	if !msg.MaxActiveInferersQuantile.IsBetweenZeroAndOneInclusive() {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "max active inferers quantile must be between 0 and 1")
+	}
+	if !msg.MaxActiveForecastersQuantile.IsBetweenZeroAndOneInclusive() {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "max active forecasters quantile must be between 0 and 1")
+	}
+	if !msg.MaxActiveReputersQuantile.IsBetweenZeroAndOneInclusive() {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "max active reputers quantile must be between 0 and 1")
+	}
```

Comment:
I'm worried you put these in the wrong spot... Perhaps I'm remembering something wrong, but I understood these parameters to set bounds on what new topics can set their parameters as (and if these parameters are adjusted down, then the minimum of the existing, immutable topic value and this global is used).

Reply:
I'm actually not sure where the existing params you have are used...

Reply:
I don't understand what you're trying to say but if you're talking about existing topics, I haven't written the migrations for this PR yet. We'll have to iterate over existing topics and add this field to them if that is what you are asking

Reply:
todo: rename better